pkg/note/note.go

```diff
@@ -0,0 +1,152 @@
+/*
+Copyright 2025 The Sigstore Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Heavily borrowed from https://gist.githubusercontent.com/AlCutter/c6c69076dc55652e2d278900ccc1a5e7/raw/aac2bafc17a8efa162bd99b4453070b724779307/ecdsa_note.go - thanks, Al
+
+package tessera
+
+import (
+	"bytes"
+	"context"
+	"crypto/ecdsa"
+	"crypto/ed25519"
+	"crypto/rsa"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/binary"
+	"fmt"
+	"strings"
+	"unicode"
+	"unicode/utf8"
+
+	"github.com/sigstore/sigstore/pkg/signature"
+	"github.com/sigstore/sigstore/pkg/signature/options"
+	"golang.org/x/mod/sumdb/note"
+)
+
+const (
+	algEd25519 = 1
+	algUndef   = 255
+	rsaID      = "RSA-PKCS#1v1.5"
```

Comment:
Can follow up later, but we should document this custom ID in a readme on log entry verification.

Reply:
https://github.com/sigstore/rekor-tiles/issues/93

Reply:
@haydentherapper what's the reasoning behind your suggestion to use `PKCS#1` as the identifier, vs `PKCS#8` or `PKIX`? It has implications on whether I use https://pkg.go.dev/crypto/x509#MarshalPKCS1PublicKey or https://pkg.go.dev/crypto/x509#MarshalPKIXPublicKey to convert the key object to bytes, I just noticed I'm using the wrong one for this identifier.

Reply:
To differentiate between the two modern RSA schemes, PSS and PKCS#1v1.5. You're right that there are two ways to encode public keys, so can we include both the scheme and public key encoding in the ID name?

Reply:
Including encoding also matches up well with how we’ve named the supported keys - https://github.com/sigstore/protobuf-specs/blob/e3f5847bde2c42348d1b99add0737ab3193f45d0/protos/sigstore_common.proto#L69